app/services/exchange_rates/create_csv_service.rb

```diff
@@ -0,0 +1,92 @@
+module ExchangeRates
+  class CreateCsvService
+    attr_reader :data
+
+    def self.call(data)
+      new(data).call
+    end
+
+    def initialize(data)
+      @data = data
+    end
+
+    def call
+      return argument_error unless valid_data?
+
+      create_csv
+    end
+
+  private
+
+    def headings
+      [
+        'Country/Territories',
+        'Currency',
+        'Currency Code',
+        'Currency Units per £1',
+        'Start date',
+        'End date',
+      ]
+    end
+
+    def create_csv
+      CSV.generate do |csv|
+        csv << headings
+
+        ordered_data = order_data
+
+        ordered_data.each do |currency_country_data|
+          csv << build_row(currency_country_data)
+        end
+      end
+    end
+
+    def build_row(currency_country_data)
+      # currency_country_data is an array with 2 elements
```

Comment:
You could use named indexes over first/last? and remove this comment?

Reply:
e.g. 

```ruby
COUNTRY_NAME_INDEX = 0
CURRENCY_RATE_INDEX = 1
```

Reply:
Happy either way, i wasnt going to leave a comment but just incase got confusing using the hash structure/array